### PR TITLE
Telemetry: Expose diagnostic counts on fetch

### PR DESCRIPTION
**Pull request by disq**
Something like this? I can't really get them to fail in this particular way so can't really test the output.

Supposed to generate `int` attributes like
`fetch.diag.error.access.aws`
`fetch.diag.warning.throttle.aws`
etc. so:
`fetch.diag.<severity lowercase>.<type lowercase>.<provider>`

And total attributes per severity/provider pair, like
`fetch.diag.error.total.aws`
`fetch.diag.warning.total.aws`
`fetch.diag.ignore.total.aws`

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -93,6 +94,46 @@ func (p ProviderFetchSummary) HasErrors() bool {
 	return false
 }
 
+func (p ProviderFetchSummary) Attributes() []attribute.KeyValue {
```
Comment (roneli) [inline pkg/client/client.go:97]: Maybe you should allow conversion of Provider into a map, and create a generic function to convert map -> attribute.KeyValue perhaps?

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM overall, had one question/discussion, before we approve and merge.

File: pkg/client/client.go (hunk 1 above)
Comment (disq) [reply pkg/client/client.go]: @roneli done

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**